.github/shared/cmd/detect-missing-rpaas-rp.js

```diff
@@ -0,0 +1,135 @@
+
+import { getChangedFiles, getChangedFilesStatuses, swagger } from "../src/changed-files.js";
+import { parseArgs } from "node:util";
+import { dirname } from "node:path";
+import { fileURLToPath } from "node:url";
+import { readdir, access, stat } from "node:fs/promises";
+import { join } from "node:path";
+
+function usage() {
+  console.log(`Usage:
+npx api-doc-preview --output <output-dir>
+
+parameters:
+  --repoWithChanges <repo directory>                         Directory checked out with PR Changes
+  --referenceFolder <azure-rest-api-specs-pr@RPSaaSMAster>   The azure-rest-api-specs-pr directory checked out at RPSaaSMaster branch.`);
+}
+
+const {
+  values: {
+    "repoWithChanges": repoWithChanges,
+    "referenceFolder": referenceFolder
+  },
+} = parseArgs({
+  options: {
+    "repoWithChanges": {
+      type: "string",
+      default: process.env.BUILD_SOURCESDIRECTORY || "",
+    },
+    "referenceFolder": {
+      type: "string",
+      default: process.env.PRIVATESPECSDIRECTORY || "",
+    },
+  },
+  allowPositionals: false,
+});
+
+let validArgs = true;
+
+
+if (!repoWithChanges) {
+  console.log(
+    `Missing required parameter --repoWithChanges. Value given: ${repoWithChanges || "<empty>"}`,
+  );
+  validArgs = false;
+}
+
+if (!referenceFolder) {
+  console.log(
+    `Missing required parameter --referenceFolder. Value given: ${referenceFolder || "<empty>"}`,
+  );
+  validArgs = false;
+}
+
+if (!validArgs) {
+  usage();
+  process.exit(1);
+}
+
+const changedFiles = await getChangedFiles({
+  cwd: repoWithChanges,
+  paths: ["specification"],
+});
+
+// Extract RP folders from changed files
+// Files will be in format: specification/<rp>/<rest-of-path>
+const changedRpFolders = new Set();
+
+for (const filePath of changedFiles) {
+  // Split the path and check if it starts with 'specification'
+  const pathParts = filePath.split('/');
+
+  if (pathParts.length >= 2 && pathParts[0] === 'specification') {
```

Comment:
We know it must start with `specification` beacause of the filter you passed to getChangedFiles().